src/Service/FinalizarVendaService.php

```diff
@@ -14,38 +17,62 @@ public function __construct(
         private CarrinhoRepository $carrinhoRepository,
         private ClienteRepository $clienteRepository,
         private ProdutoRepository $produtoRepository,
+        private ItemRepository $itemRepository,
     ) {
     }
-
-    public function execute(int $id): Carrinho
+    
+    public function execute(int $clienteId, array $produtos): Carrinho
     {
+        $cliente = $this->clienteRepository->find($clienteId);
+        $carrinho = $this->carrinhoRepository->buscarUltimoCarrinhoPendente(["cliente" => $cliente]);
 
-        $cliente = $this->clienteRepository->find($id);
-        $carrinho = $this->carrinhoRepository->findOneBy(["cliente" => $cliente]);
-    
-        if ($carrinho->getItems()->isEmpty()) {
-         throw new BadRequestHttpException('O carrinho não contém produtos.');
-        }
-    
-        if ($carrinho->getStatus() !== StatusEnum::aberto) {
-            throw new BadRequestHttpException('Não é possível finalizar um carrinho que não está pendente.');
+        if(null === $carrinho){
+            throw new \Exception("O cliente não possui nenhum carrinho pendente a ser finalizado!!");
         }
-   
-        foreach ($carrinho->getItems() as $item) {
-           $produto = $item->getProduto();
+       
+        foreach($produtos as $produto){
+            $produtoId = $produto['id']; 
+            $quantidade = $produto['quantidade']; 
+            
+            $produtoEncontrado = $this->produtoRepository->find($produtoId);
+            $quantidadeDisponivel = $produtoEncontrado->getQuantidadeDisponivel();
 
-           if ($produto->getQuantidadeDisponivel() === 0) {
-               throw new BadRequestHttpException('O produto ' . $produto->getNome() . ' está fora de estoque.');
+            if (!$produtoEncontrado) {
+                throw new \Exception("Produto não encontrado");
             }
```

Comment:
move isso para antes da linha 38, no caso deixar entre a busca e a variavel quantidadeDisponivel...
se o produto não for encontrado não tem como pegar o getQuantidadeDisponivel 🚀

Reply:
beleza